hipcheck/src/plugin/manager.rs

```diff
@@ -0,0 +1,122 @@
+use crate::hipcheck::plugin_client::PluginClient;
+use crate::plugin::{HcPluginClient, Plugin, PluginContext};
+use crate::{hc_error, Result, F64};
+use rand::Rng;
+use std::collections::HashSet;
+use std::ops::Range;
+use std::process::Command;
+use tokio::time::{sleep_until, Duration, Instant};
+
+pub struct PluginExecutor {
+	max_spawn_attempts: usize,
+	max_conn_attempts: usize,
+	port_range: Range<u16>,
+	backoff_interval: Duration,
+	jitter_pct: u8,
+	est_ports: HashSet<u16>,
```

Comment:
Small preference thing: I'd prefer not to abbreviate to "est" or at least to spell out in a comment what it's a shorthand for.

Reply:
Got it. I can update it if you want, but I'll also say that I have already removed this field in the next PR i intend to merge. For the purposes of using `join_all()` to initialize all plugins concurrently in async fashion, the Executor can't have mutable state

Reply:
Okay cool, if it's getting removed that's fine. As far as mutexes, Tokio does have an async mutex in it.

Reply:
Also, we may want to look into: https://marabos.nl/atomics/

Reply:
On the topic of abbreviations, I wasn't sure about `pct` for "percent". I only became sure that it stood for "percent" after finding how the value was used, and in particular that it was divided by 100 at one point.

But maybe `pct` is a well established abbreviation that I just haven't seen much before.

Reply:
Fair, I'll update that one.